WordPress/Classes/Services/AccountService.m

```diff
@@ -35,22 +35,20 @@ @implementation AccountService
  */
 - (WPAccount *)defaultWordPressComAccount
 {
-    NSString *uuid = [[NSUserDefaults standardUserDefaults] stringForKey:DefaultDotcomAccountUUIDDefaultsKey];
-    if (uuid.length == 0) {
+    NSURL *myObjectURL = [NSURL URLWithString:[[NSUserDefaults standardUserDefaults] objectForKey:DefaultDotcomAccountIDDefaultsKey]];
+    NSManagedObjectID *myObjectID = [self.managedObjectContext.persistentStoreCoordinator managedObjectIDForURIRepresentation:myObjectURL];
+
+    if (!myObjectID) {
         return nil;
     }
 
-    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Account"];
-    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"uuid == %@", uuid];
-    fetchRequest.predicate = predicate;
-    
     NSError *error = nil;
-    NSArray *fetchedObjects = [self.managedObjectContext executeFetchRequest:fetchRequest error:&error];
+    WPAccount *foundAccount = [self.managedObjectContext existingObjectWithID:myObjectID error:&error];
```

Comment:
We're gonna need to revert to the previous version of this method (based on `uuid`). 

**NSManagedObjectID** do change after a heavyweight migration ([we've seen that issue in the past](https://astralbodi.es/2014/10/22/core-data-object-ids-can-change/)), and it's a potential problem source.